packages/grafana-runtime/src/services/live.ts

```diff
@@ -0,0 +1,72 @@
+import { PartialObserver, Unsubscribable } from 'rxjs';
+
+/**
+ * @experimental
+ */
+export interface ChannelHandler<T = any> {
+  /**
+   * Process the raw message from the server before broadcasting it
+   * to all subscribeers on this channel
+   */
+  onPublish(msg: any): T;
+}
+
+// export interface SubscriptionEvents {
+//   publish?: (ctx: PublicationContext) => void;
+//   join?: (ctx: JoinLeaveContext) => void;
+//   leave?: (ctx: JoinLeaveContext) => void;
+//   subscribe?: (ctx: SubscribeSuccessContext) => void;
+//   error?: (ctx: SubscribeErrorContext) => void;
+//   unsubscribe?: (ctx: UnsubscribeContext) => void;
+// }
+
+/**
+ * @experimental
+ */
+export interface GrafanaLiveSrv {
+  /**
+   * Is the server currently connected
+   */
+  isConnected(): boolean;
+
+  /**
+   * Listen for changes to the connection state
+   */
+  connection(observer: PartialObserver<boolean>): Unsubscribable;
+
+  /**
+   * Configure a channel with the given setup
+   */
+  initChannel<T>(channel: string, handler: ChannelHandler<T>): void;
+
+  /**
+   * Subscribe to activity on a given channel
+   */
+  subscribe<T>(channel: string, observer: PartialObserver<T>): Unsubscribable;
```

Comment:
Is there a reason for passing in an observer here? instead of returning an Observable? returning an Observable (I think) will make this easier to use as you can use it together with rxjs operators etc

Reply:
updated to return Observable -- yes this makes more sense :+1: